rfc/system/0008-community-plugins.md

```diff
@@ -0,0 +1,48 @@
+# Management of community-contributed plugins
+
+**Authors:**
+
+- @davidmirror-ops
+
+
+## 1 Executive Summary
+
+The Flyte community as a self-governed and productive collective of individuals, values contributions. This proposal aims to discuss the process community contributors should follow to submit a new `flytekit` plugin, with special attention to mechanisms that ensure stability and maintainability of core flyte code.
+
+## 2 Motivation
+
+- With the current "in-tree" approach, plugins developed by the community land on the `flytekit` repo ([example](https://github.com/flyteorg/flytekit/pull/2537)). It results in Flyte maintainers having to take care of CI test failures due to plugin code or flytekit updates incompatible with plugin code, etc.
+
+- The goal is to agree on a process for contributors to follow when submitting new integrations in a "out-of-tree" way that clearly communicates that it is a community-contributed -and then- community-supported integration.
+
+## 3 Proposed Implementation
+
+- Create a `community` folder under `flytekit/plugins` and publish the plugins in that folder as separate `pypi` packages.
+- Configure CI to only run tests on `plugins/community` when there are changes.
+- Keep releasing plugins alongside flytekit.
+- Update [CODEOWNERS](https://github.com/flyteorg/flytekit/blob/master/CODEOWNERS), enabling contributors to merge community plugins (it requires granting Write permissions to contributors on `plugins/community`).
+
```

Comment:
What do you think of explicitly marking them as community maintained in the import via `import flytekitplugins.contrib.x`?

Reply:
How about `import flytekitplugins.community.x`?
like https://pypi.org/project/langchain-community/

Reply:
I like it!

Reply:
I believe we will have `flytekitplugins.experimental` and `flytekitplugins.partners` in the future.

Reply:
`contrib` has been used e.g. by tensorflow back in the day.

Reply:
yup. I'm okay with both

Reply:
I agree with using `contrib` as it communicates even more broadly, beyond the location in the repo

Reply:
cc @eapolinario

Reply:
+1 to `contrib`.